### PR TITLE
Don't load commercial JS for users with poor device connectivity

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- refactors the commercial JS loading (it was getting a bit hard to follow)
- prevents the commercial JS loading for users identified as having poor device connectivity (PDC)

in reality, the PDC cohort is currently a server-side experiment test group whose behaviour @guardian/open-journalism want to observe. 

we'll be adding/removing features to see if/how they influence behaviour, from a perf point of view.

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes https://github.com/guardian/dotcom-rendering/pull/7340

### CODE REVIEW

File: static/src/javascripts/boot.js [hunk 1]
```diff
@@ -151,28 +151,34 @@ const go = () => {
...
-                ? loadScript(config.get('page.commercialBundleUrl')).then(
-                    () => fakeBootCommercial,
-                )
-                : import(
-                    /* webpackChunkName: "commercial" */
-                    'bootstraps/commercial-legacy'
-                );
+        // Start downloading these ASAP
 
+		const fetchCommercial = () => {
```
Comment (mxdvl) [inline static/src/javascripts/boot.js:156]: Why has this become a method/function rather than a constant?

**Review by mxdvl — APPROVED** [review]
Looks good to me, but would be good to deploy in CODE and get @guardian/commercial-dev to kick the tyres!

Curious about the implications of going from a promise to a function that returns a promise. I’m assuming it’s quite minimal.

File: static/src/javascripts/boot.js (hunk 1 above)
Comment (sndrs) [reply static/src/javascripts/boot.js:156]: in part to avoid having to do things like `(markTime('commercial request'), commercialBundle())`, and also having to disable `no-nested-ternary`. 

allows you to not trigger anything until you know you'll need it while trying to make it as legible as possible

File: static/src/javascripts/boot.js [hunk 2]
```diff
@@ -181,7 +187,7 @@ const go = () => {
             : Promise.resolve({ bootEnhanced: () => { } });
 
         Promise.all([
-            fetchCommercial.then(({ bootCommercial }) => {
+            fetchCommercial().then(({ bootCommercial }) => {
```
Comment (arelra) [inline static/src/javascripts/boot.js:190]: Just acknowledging the noop pattern is a bit convoluted. It seems to only exist for commercial-legacy now. I'll add this to a clean up ticket.

**Review by arelra — APPROVED** [review]
